OpenXmlPowerTools/DocumentBuilder.cs

```diff
@@ -174,6 +174,11 @@ public Source(WmlDocument source, int start, int count, string insertId)
         }
     }
 
+    public class DocumentBuilderSettings
+    {
+        public HashSet<string> CustomXmlGuidList = null;
```

Comment:
I'd still suggest hiding that it's a hashset and exposing it as a readonly ICollection. Fields are against the framework guidelines and should be a property. I'd suggest the following:

```csharp
public class DocumentBuilderSettings
{
  public ICollection<string> CustomXmlGuids { get; } = new HashSet<string>(StringComparer.Ordinal);
}
```

I think this would be better because:

- You ensure the comparer is set correctly - I'm not sure about the semantics exactly, but you definitely want either `Ordinal` or `OrdinalIgnoreCase`
- You can potentially change it out later without a breaking change
- Also having the name suffixed with `List` isn't really necessary and seems cleaner to just pluralize it

Reply:
Also, thinking about it, why not have it as `Guid`? That way it's strongly typed and you don't even need to think about comparing issues with strings (and it's only 4 bytes vs 20 or something)